railties/lib/rails/application/bootstrap.rb

```diff
@@ -73,7 +73,7 @@ module Bootstrap
 
       initializer :initialize_event_reporter, group: :all do
         Rails.event.raise_on_error = config.consider_all_requests_local
-        Rails.event.debug_mode = Rails.env.development?
+        Rails.event.debug_mode = Rails.env.development? || config.log_level.to_s == "debug"
```

Comment:
We should be able to do what you'd mentioned previously and drop the `Rails.env.development?` (the log level will default to debug in development)
```suggestion
        Rails.event.debug_mode = config.log_level.to_s == "debug"
```

Reply:
@adrianna-chang-shopify done!